gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -169,9 +175,8 @@ public void run() {
         LOG.error("Failed to close all open resources", t);
       }
 
-      long endTime = System.currentTimeMillis();
-      this.taskState.setEndTime(endTime);
-      this.taskState.setTaskDuration(endTime - startTime);
+      this.taskState.setTaskDuration(TimeUnit.NANOSECONDS.toMillis(taskTimerContext.stop()));
+      this.taskState.setEndTime(System.currentTimeMillis());
```

Comment:
Should the TaskState object be tracking task level metrics. I would assume the JobMetrics's MetricsRegistry object could be used for this purpose? I ask because if we are going to collect task and map level metrics, then why not track all the metrics using the MetricsRegistry rather than having some metrics in the MetricsRegistry, and some other in the TaskState.